src/vscode-azurefunctions.api.d.ts

```diff
@@ -9,12 +9,24 @@ export interface AzureFunctionsExtensionApi {
     revealTreeItem(resourceId: string): Promise<void>;
 
     createFunction(options: ICreateFunctionOptions): Promise<void>;
+    downloadAppSettings(client: ISimpleAppSettingsClient): Promise<void>;
+    uploadAppSettings(client: ISimpleAppSettingsClient): Promise<void>;
 }
 
 export type ProjectLanguage = 'JavaScript' | 'TypeScript' | 'C#' | 'Python' | 'PowerShell' | 'Java';
-
 export type ProjectVersion = '~1' | '~2' | '~3';
 
+export interface ISimpleAppSettingsClient {
```

Comment:
Let's rename `ISimpleAppSettingsClient` to `IAppSettingsClient`

I feel like the only reason you didn't do that already is because of `IAppSettingsClient` in vscode-azureappservice, but I don't think that's a good enough reason to pick a worse name here. They both have a legitimate reason to be called `IAppSettingsClient` and we can work around any problems arising from that.

Reply:
Yeah, that is pretty much the reason why.  That and this isn't really a "complete" `AppSettingsClient` since you wouldn't be able to use it with an `AppSettingsTreeNode`.  But I'm okay with changing it, just didn't want to cause confusion.